src/vs/workbench/services/configuration/browser/configuration.ts

```diff
@@ -127,15 +128,15 @@ export class UserConfiguration extends Disposable {
 	get hasTasksLoaded(): boolean { return this.userConfiguration.value instanceof FileServiceBasedConfiguration; }
 
 	constructor(
-		private readonly userSettingsResource: URI,
+		private readonly userDataProfile: IUserDataProfile,
 		scopes: ConfigurationScope[] | undefined,
 		private readonly fileService: IFileService,
 		private readonly uriIdentityService: IUriIdentityService,
 		private readonly logService: ILogService,
 	) {
 		super();
 		this.configurationParseOptions = { scopes, skipRestricted: false };
-		this.userConfiguration.value = new UserSettings(this.userSettingsResource, scopes, uriIdentityService.extUri, this.fileService);
+		this.userConfiguration.value = new UserSettings(this.userDataProfile.settingsResource, scopes, uriIdentityService.extUri, this.fileService);
```

Comment:
How do we support reading & writing application scoped settings via default profile explicitly?

Reply:
Will be handled in further PRs